tests/steps/other_v2_steps.py

```diff
@@ -1530,6 +1531,111 @@ def power_pack_simulation_should_have_extra_budget(context, budget):
     )
 
 
+@then(
+    'I allow exec trace options "{options:MaybeString}" on that simulate request.'
+)
+def exec_trace_config_in_simulation(context, options: str):
+    option_list = options.split(",")
+    context.simulate_request.exec_trace_config = SimulateTraceConfig(
+        enable=True,
+        stack_change="stack" in option_list,
+        scratch_change="scratch" in option_list,
+    )
+
+
+@then(
+    '{unit_index}th unit in the "{trace_type}" trace at txn-groups path "{group_path}" should add value "{stack_addition:MaybeString}" to stack, pop {pop_count} values from stack, write value "{scratch_var:MaybeString}" to scratch slot "{scratch_index:MaybeString}".'
+)
+def step_impl(
+    context,
+    unit_index,
+    trace_type,
+    group_path,
+    stack_addition: str,
+    pop_count,
+    scratch_var,
+    scratch_index,
+):
+    assert context.atomic_transaction_composer_return
+    assert context.atomic_transaction_composer_return.simulate_response
+
+    simulation_response = (
+        context.atomic_transaction_composer_return.simulate_response
+    )
+    assert "txn-groups" in simulation_response
+
+    assert simulation_response["txn-groups"]
+
+    group_path = list(map(int, group_path.split(",")))
+    traces = simulation_response["txn-groups"][0]["txn-results"][
+        group_path[0]
+    ]["exec-trace"]
+    assert traces
+
+    for i in range(1, len(group_path)):
+        traces = traces["inner-trace"][group_path[i]]
+        assert traces
+
+    trace = []
+    if trace_type == "approval":
+        trace = traces["approval-program-trace"]
+    elif trace_type == "clearState":
+        trace = traces["clear-state-program-trace"]
+    elif trace_type == "logic":
+        trace = traces["logic-sig-trace"]
+
+    assert trace
+
+    unit_index = int(unit_index)
+    unit = trace[unit_index]
+
+    def compare_avm_value_with_string_literal(string_literal, avm_value):
+        [avm_type, value] = string_literal.split(":")
+        assert avm_type in ["uint64", "bytes"]
+        assert "type" in avm_value
+        if avm_type == "uint64":
+            assert avm_value["type"] == 2
+            if int(value) > 0:
+                assert avm_value["uint"] == int(value)
+            else:
+                assert "uint" not in avm_value
+        elif avm_value == "bytes":
+            assert avm_value["type"] == 1
+            if len(value) > 0:
+                assert avm_value["bytes"] == base64.b32decode(bytearray(value))
```

Comment:
shouldn't this be `b64decode`?

Reply:
good catch, my typo, fixed in [pr comment response](https://github.com/algorand/py-algorand-sdk/pull/505/commits/9c9baade970c1cdae430a231df7cdd8279464fb9)